packages/browser/src/login/oidc/redirectHandler/AuthCodeRedirectHandler.ts

```diff
@@ -71,46 +113,60 @@ export default class AuthCodeRedirectHandler implements IRedirectHandler {
       }
     )) as string;
 
-    let signinResponse;
-    try {
-      signinResponse = await new OidcClient({
-        // TODO: We should look at the various interfaces being used for storage,
-        //  i.e. between oidc-client-js (WebStorageStoreState), localStorage
-        //  (which has an interface Storage), and our own proprietary interface
-        //  IStorage - i.e. we should really just be using the browser Web Storage
-        //  API, e.g. "stateStore: window.localStorage,".
+    const isDpop =
+      (await this.storageUtility.getForUser(storedSessionId, "dpop")) ===
+      "true";
 
-        // We are instantiating a new instance here, so the only value we need to
-        // explicitly provide is the response mode (default otherwise will look
-        // for a hash '#' fragment!).
-        // eslint-disable-next-line @typescript-eslint/camelcase
-        response_mode: "query",
-        // The userinfo endpoint on NSS fails, so disable this for now
-        // Note that in Solid, information should be retrieved from the
-        // profile referenced by the WebId.
-        loadUserInfo: false,
-      }).processSigninResponse(redirectUrl.toString());
-    } catch (err) {
-      throw new Error(
-        `Problem handling Auth Code Grant (Flow) redirect - URL [${redirectUrl}]: ${err}`
-      );
-    }
+    let tokens: TokenEndpointResponse | TokenEndpointDpopResponse;
+    let authFetch: typeof fetch;
+
+    if (isDpop) {
+      // Since we throw if not found, the type assertion is okay
+      const issuer = (await this.storageUtility.getForUser(
+        storedSessionId,
+        "issuer",
+        { errorIfNull: true }
+      )) as string;
```

Comment:
Here too seems like `getForUser` could have type overloads that indicate that it returns a `string` if `errorIfNull` is set to `true`.

Reply:
That is a great idea. However, I just gave it a try, and the fact that the functions to overload are class methods rather than regular functions give a rather strange behaviour, as if only the latest type applied. I created a ticket to follow this up.

Reply:
Ah that's weird. Just to make sure, when you add overloads, you also need to add one for the actual function definition, as that gets overridden.

So for example:

```
function aFunction(paramA: true): string;
function aFunction(paramA: boolean): string | null;
function aFunction(paramA: boolean): string | null {
  /* Return a string on true, or either when it's not. */
}
```

Reply:
Ah, I wasn't doing that, so that may have caused the issue (Did I shamelessly blame the tool because I did not know how to use it properly ? Maybe.) But even writing
```
async getForUser(
    userId: string,
    key: string,
    options: { errorIfNull: true; secure?: boolean }
  ): Promise<string>;
  async getForUser(
    userId: string,
    key: string,
    options?: { errorIfNull?: boolean; secure?: boolean }
  ): Promise<string | undefined>;
  async getForUser(
    userId: string,
    key: string,
    options?: { errorIfNull?: boolean; secure?: boolean }
  ): Promise<string | undefined>
```
the type assertion is required, otherwise the TS compiler complains.

Reply:
Hmm, might be because the nested object parameter is too hard for TS to pattern match? :shrug: